gherkin/java/src/main/java/io/cucumber/gherkin/pickles/PickleCompiler.java

```diff
@@ -44,10 +44,10 @@ public PickleCompiler(IdGenerator idGenerator) {
 
     public List<Pickle> compile(GherkinDocument gherkinDocument, String uri) {
         List<Pickle> pickles = new ArrayList<>();
-        Feature feature = gherkinDocument.getFeature();
-        if (feature == null) {
+        if (!gherkinDocument.getFeature().isPresent()) {
             return pickles;
         }
+        Feature feature = gherkinDocument.getFeature().get();
```

Comment:
This is `.isPresent` followed by `.get` that can be replaced with `.map`. There are few more in this method.